getting_started/workflow/best_practices/autoloads_versus_internal_nodes.rst

```diff
@@ -1,103 +1,95 @@
 .. _doc_autoloads_versus_internal_nodes:
 
-Autoloads versus internal nodes
+Autoloads versus regular nodes
+==============================
+
+Godot offers a feature to automatically load nodes at the root of your project,
+making allowing you to access them globally, that can fulfill the role of a
+Singleton: :ref:`doc_singletons_autoload`. These auto-loaded nodes are not freed
+when you change the scene from code with ``SceneTree.change_scene``.
+
+In this guide, you will learn when to use the Autoload feature, and techniques
+you can use to avoid it.
+
+The cutting audio issue
+=======================
+
+Other engines can encourage the use of creating manager classes, singletons that
+organize a lot of functionality into a globally accessible object. Godot offers
+many ways to avoid global state thanks to the node tree and signals.
+
+For example, let's say we are building a platformer and want to collect coins
+that play a sound effect. There's a node for that: the :ref:`AudioStreamPlayer
+<class_AudioStreamPlayer>`. But if we if call the ``AudioStreamPlayer`` while it
+is already playing a sound, the new sound interrupts the first.
+
+A solution is to code a global, auto-loaded sound manager class. It generates a
+pool of ``AudioStreamPlayer`` that cycle through as each new request for sound
+effects comes in. Say we call that class ``Sound``, you can use it from anywhere
+in your project by calling ``Sound.play("coin_pickup.ogg")``. This solves the
+problem in the short term but causes more problems:
+
+1. **Global state**: one object is now responsible for all objects' data. If the
+   ``Sound`` class has errors or doesn't have an AudioStreamPlayer available,
+   all the nodes calling it can break.
+
+2. **Global access**: now that any object can call ``Sound.play(sound_path)``
+   from anywhere, there's no longer an easy way to find the source of a bug.
+
+3. **Global resource allocation**: with a pool of ``AudioStreamPlayer`` nodes
+   stored from the start, you can either have too few and face bugs, or too many
+   and use more memory than you need.
+
+Contrast this with each scene keeping as many ``AudioStreamPlayer`` nodes as it
+needs within itself and all these problems go away:
+
+1. Each scene manages its own state information. If there is a problem with the
+   data, it will only cause issues in that one scene.
+
+2. Each scene accesses only its own nodes. Now, if there is
+   a bug, it's easy to find which node is at fault.
+
+3. Each scene allocates exactly the amount of resources it needs.
+
+Managing shared functionality or data
+=====================================
+
+Another reason to use an Autoload can be that you want to reuse the same method
+or data across many scenes.
+
+In the case of functions, you can create a new type of ``Node`` that provides
+that feature for an individual scene using the :ref:`class_name
+<doc_scripting_continued_class_name>` keyword in GDScript.
+
```

Comment:
Another thing that is important to discuss here for "in the case of functions" is the difference between pure and impure functions and how one can create a library of independently testable static functions within script classes. I elaborate more on this content in [section `1.b.i` of this reddit comment](https://www.reddit.com/r/godot/comments/dhjohq/best_ways_to_implement_systems_with_many/f3om27g/). If you would like, you could add a tidbit here in a second paragraph, before the "when it comes to data" section.

Reply:
Yup definitely, that's actually mostly how we use them - function libraries.

Reply:
Thinking of this again, I don't think this discussion is a good fit for a guide about "when and how to use auto-loaded nodes in Godot" because you can use a gdscript file that doesn't extend anything to create a library of static functions, i.e. with class_name